js/clipboardToggleAll.js

```diff
@@ -0,0 +1,39 @@
+(function ($)
+  {
+    // Toggle buttons
+    var clickClipButton = function (node, selectAll) {
+      $button = $(node);
+
+      // if 'none' was chosen, click any button that has 'added' on
+      if (selectAll == false && $button.hasClass('added')) {
+        $button.click();
+      }
+      // if 'all' was chosen, don't click the button if 'added' is on
+      else if (selectAll == true && !$button.hasClass('added')) {
+        $button.click();
+      }
+    };
+
+    // Convenience wrapper in jQuery
+    $.fn.clickClipButton = function(clicked) {
+      this.each(function () {
+        clickClipButton.apply(null, [this, clicked]);
+      });
+    };
+
+    // get all affected clipboard buttons and select all or none based on
+    // item clicked.
+    $(document).ready(function() {
+
+      var $area = $('#clipboardButtonNode');
+
+      $area.find('.all').on('click', function (event) {
+        $area.find('button').clickClipButton(true);
```

Comment:
Both callbacks should probably call `event.preventDefault()` to cancel the browser's default event when you click on the `a` element (go to `#`).